components/event-sources/reconciler/httpsource/controller.go

```diff
@@ -57,11 +65,27 @@ func init() {
 	utilruntime.Must(sourcesscheme.AddToScheme(scheme.Scheme))
 }
 
+func NewControllerWrapper(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
```

Comment:
The introduction of this wrapper looks unnecessary to me. You could just call some initialization function _inside_ `NewController()` instead of making `NewController()` a dependency of yet another constructor.

Reply:
Right, I refactored by using injection clients for Istio Auth Policy.